[MagnificentCreature] iP

MagnificentCreature opened the pull request:
# Forine 

> “Beep beep.” – Sheep ([source](https://example.com))

Forine **frees** you.

* _test-based_
* _learning is hard_
* _FAST_ ~~SUPER FAST~~ to use

_All you need to do is_,

1. download it from [here](https://github.com/MagnificentCreature/ip).
2. `double-click` it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

## Features:

- [x] Managing words
- [x] Managing boxes with words on its right
- [ ] other things

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

MagnificentCreature committed:
First commit

MagnificentCreature committed:
First commit

MagnificentCreature committed:
Merge branch 'master' of https://github.com/MagnificentCreature/ip

MagnificentCreature committed:
Delete Duke.class

MagnificentCreature committed:
Level 1

MagnificentCreature committed:
Level-2

MagnificentCreature committed:
Level-3

MagnificentCreature committed:
Level-4

MagnificentCreature committed:
A-TextUiTesting

MagnificentCreature committed:
Level-5

MagnificentCreature committed:
Level--6

MagnificentCreature committed:
A-Enum

MagnificentCreature committed:
small changes

MagnificentCreature committed:
Formatting changes

MagnificentCreature committed:
Implement functionality to save and load tasks

MagnificentCreature committed:
Merge pull request #1 from MagnificentCreature:branch-Level-7

Branch-Level-7

MagnificentCreature committed:
Alot of major refactoring

MagnificentCreature committed:
Major code refactor

MagnificentCreature committed:
Add more error messages
Add more OOP

MagnificentCreature committed:
bug fixes

MagnificentCreature committed:
Merge pull request #2 from MagnificentCreature:branch-Level-8

Branch-Level-8

MagnificentCreature committed:
Rename messages to Ui

MagnificentCreature committed:
Created Gradle files

MagnificentCreature committed:
Stable gradle project

MagnificentCreature committed:
Rename Messages to Ui

MagnificentCreature committed:
Merge branch 'master' into A-Gradle

MagnificentCreature committed:
Merge pull request #4 from MagnificentCreature/A-Gradle

A-Gradle

MagnificentCreature committed:
A-JUnit

MagnificentCreature committed:
A-Jar

MagnificentCreature committed:
Add Jdocs for most public methods and fields

MagnificentCreature committed:
Enforced coding standards

MagnificentCreature committed:
Level-9

MagnificentCreature committed:
improve findTask to search by word

MagnificentCreature committed:
Merge pull request #5 from MagnificentCreature/A-JavaDoc

Add Jdocs for most public methods and fields

MagnificentCreature committed:
Merge branch 'A-CodingStandard'

MagnificentCreature committed:
Merge branch 'Level-9'

MagnificentCreature committed:
Bug fixes and minor improvements

Users can now put "|" in the description of a task without causing bugs.
Allow keying in of dates in the format YYYY-M-D for deadline and event tasks, that have single digit months and days.

MagnificentCreature committed:
A-CheckStyle

MagnificentCreature committed:
Comment changes

MagnificentCreature committed:
Merge pull request #8 from MagnificentCreature/A-CheckStyle

Comment changes

yihfei commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws InvalidIndexException if the index provided is invalid
+     * @throws InvalidTimeFormatException if the time provided is invalid
+     */
+    public Command dispatch(String input)
```
I like the command class and how it's organised! Perhaps could reconsider renaming this method to parse as I couldn't tell what dispatch meant until I read the javadoc comment

MagnificentCreature replied:
I am considering it, but I already have another method called parse. That parse will be overridden for each command depending on how the command is meant to be parsed. Dispatch is used to run that parse.

That  being the case I could not find a comprehensive way to fuse the two and left it as such

yihfei commented on src/main/java/duke/parser/IndexCommandParser.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Override
+    protected Command createCommand(Matcher matcher)
```
I like the use of cases here!

yihfei commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,21 @@
+     * @throws MissingDescriptionException If the description is missing.
+     * @throws IncorrectCommandFormatException If the command is in the wrong format.
+     */
+    void execute(TaskListStorage taskListStorage)
```
Is there any reason why there wasn't a access modifier here

MagnificentCreature replied:
Oops, fixed

yihfei commented on src/main/java/duke/parser/EventCommandParser.java:
```diff
@@ -0,0 +1,73 @@
+    //CHECKSTYLE.ON: LineLength
+
+    @Override
+    protected void validate(Matcher matcher)
```
I like how comprehensive this method is :))

yihfei reviewed (commented):
Overall, the code looks extremely clean and readable. Really nothing much to change here!

dylkaw commented on src/main/java/duke/TaskListStorage.java:
```diff
@@ -0,0 +1,258 @@
+            boolean isDone = taskInfo[1].equals("1");
+            String taskDescription = taskInfo[2].replace("\\|", "|"); //Unescape
+            try {
+                switch (taskType) {
```
Is there a reason why you did not use an enum for the task types?

MagnificentCreature replied:
This is the rawest form of parsing the inputs from the file, as such enums are not necessary. Creating an enum will mean I must then associate the enum with the letter I'm using to save to the file which seems like extra space and computational work for not much increase in understanding/readability of the code. 

Without enums, it should be easy to see D corresponds to creating a deadlineTask since the following line is adding a deadline task to the tasklist.

The only time I might consider using enums is if the task types are a considerably high number that 

1)I cant just use simple alphabets to represent
2)I might need to map what I write in the task file to its own unique task, ie D1, D2 etc.

dylkaw commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+         * @param commandString the string representation of the command
+         * @return the command with the specified string representation
+         */
+        public static Commands valueOfLabel(String commandString) {
```
Maybe this can be named using a verb?

MagnificentCreature replied:
Thanks for the suggestion! I will change this

dylkaw reviewed (commented):
Overall well organised and easy to read!

MagnificentCreature committed:
Added Level-10 GUI

MagnificentCreature committed:
Merge pull request #9 from MagnificentCreature:Level-10

Merge pull request #8 from MagnificentCreature/A-CheckStyle

MagnificentCreature committed:
Improve code quality

extractTaskDescription: changed header comment to follow format
specified in the coding standard, in particular, the phrasing of the
overview statement started with the words "The following methods are..."
This redundant statement has been removed.

Other segments of dead code from Duke and DialogBox have also been
removed.

MagnificentCreature committed:
Add assertions to the code

Used assert feature to document important assumptions that should hold
at various points in the code.
These include:
-Assering that the filepath for tasks data is valid
-Asserting that the task is valid when reading from file

MagnificentCreature committed:
Merge pull request #11 from MagnificentCreature:branch-A-CodeQuality

Improve code quality

MagnificentCreature committed:
Merge pull request #12 from MagnificentCreature:A-Assertions

A-Assertions

MagnificentCreature committed:
B-Reminders

MagnificentCreature committed:
Add scheduler to constantly update mutex

MagnificentCreature committed:
minor changes

MagnificentCreature committed:
Checkstyle fixes

MagnificentCreature committed:
update ug

MagnificentCreature committed:
added Ui

MagnificentCreature committed:
add javadocs for duke class

MagnificentCreature committed:
A-Release

MagnificentCreature committed:
Fix javafx version
